tensorflow_probability/python/distributions/empirical.py

```diff
@@ -0,0 +1,369 @@
+# Copyright 2018 The TensorFlow Probability Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ============================================================================
+"""The Empirical distribution class."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import tensorflow as tf
+from tensorflow.python.ops import gen_array_ops
+from tensorflow_probability.python.distributions import distribution
+from tensorflow_probability.python.internal import distribution_util as util
+from tensorflow_probability.python.internal import reparameterization
+
+__all__ = [
+    "Empirical",
+    "VectorEmpirical"
+]
+
+
+def _broadcast_event_and_samples(event, samples, event_ndims):
+  """Broadcasts the event or samples."""
```

Comment:
At first glance it's not easy to understand what this method does. Can you add an example to the docstring?